src/modeler/AutoRest.Swagger/Validation/Extensions/XmsClientNamePropertyValidation.cs

```diff
@@ -42,26 +42,22 @@ public class XmsClientNameValidation : TypedRule<Dictionary<string, Schema>>
         /// <summary>
         /// Validates if the name of property and x-ms-client-name(if exists) does not match.
         /// </summary>
-        /// <param name="definitions">Operation Definition to validate</param>
-        /// <returns>true if the validation succeeds. false otherwise.</returns>
-        public override bool IsValid(Dictionary<string, Schema> definitions)
+        /// <param name="properties">Properties.</param>
+        /// <param name="context">Rule context.</param>
+        /// <returns></returns>
+        public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<string, Schema> properties, RuleContext context)
         {
-            foreach (string key in definitions.Keys)
+            foreach(KeyValuePair<string, Schema> property in properties)
             {
-                Schema schema = definitions.GetValueOrNull(key);
-                if(schema != null)
+                if (property.Value?.Extensions?.Count != 0)
```

Comment:
But you can have a null value and the extensions property can also be a null (hence the null-coalesce operator)